### PR TITLE
feat: build docs folder and deploy to GitHub Pages 

### LINKED ISSUE #64: Build docs to GitHub Pages with AsciiDoctor
**Description**

To view the docs on GitHub Pages with be better DX. This can include diagrams and generate a PDF later as required

https://asciidoctor.org

### PR DESCRIPTION
**Pull request by eddiejaoude**
Fixes #64

#### Proposed Changes

- Build docs folder using AsciiDoctor
- Deploy docs to GitHub Pages

#### Screenshots

Screenshot 2021-03-09 at 23 18 27

https://eddiejaoude.github.io/stargate/

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: Docs
+on:
+  push:
+    branches:
+      - main
+      - issue-64
```
Comment (andrew-cunliffe) [inline .github/workflows/docs.yml:6]: should probably remove this? only want to build docs on the main branch no?

File: package.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
   "scripts": {
     "build": "tsc",
     "lint": "eslint src",
+    "docs": "asciidoctor -D build/ docs/index.adoc"
```
Comment (andrew-cunliffe) [inline package.json]: needs a comma on the end of the line? (I don't see one) not valid json like this?

**Review by andrew-cunliffe — COMMENTED** [review]
Looks good, two small comments

File: package.json (hunk 2 above)
Comment (eddiejaoude) [reply package.json]: thanks, doh bad merge conflict 🤦‍♂️ - CI failed then I realised the issue

File: .github/workflows/docs.yml (hunk 1 above)
Comment (eddiejaoude) [reply .github/workflows/docs.yml:6]: Yep it needs to go 👍

**Pull request merged**